From 4948e146c5f765c09f175a1fef1624cc625ba1b9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Franti=C5=A1ek=20Zatloukal?= <fzatlouk@redhat.com>
Date: Sun, 11 Aug 2024 00:52:07 -0400
Subject: [PATCH] Switch wlroots to the new pc filename

---
 src/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/meson.build b/src/meson.build
index da811e8..2654d80 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -25,7 +25,7 @@ stb_dep = dependency('stb')
 avif_dep = dependency('libavif', version: '>=1.0.0', required: get_option('avif_screenshots'))
 
 wlroots_dep = dependency(
-  'wlroots',
+  'wlroots-0.18',
   version: ['>= 0.18.0', '< 0.19.0'],
   fallback: ['wlroots', 'wlroots'],
   default_options: ['default_library=static', 'examples=false', 'xwayland=enabled', 'backends=libinput', 'renderers=[]', 'allocators=[]', 'session=enabled'],
-- 
2.45.2

